pythonlib/amlrealtimeai/resnet50/utils.py

```diff
@@ -5,21 +5,147 @@
 def preprocess_array(in_images):
     return tf.map_fn(_decode, in_images, tf.float32, name="map_normalize_jpeg")
 
+def _decode(tensor):
+    return tf.squeeze(_preprocess_tensor(tensor))
+
+    
 def _preprocess_tensor(tensor):
-    jpeg_decoded = tf.image.decode_png(tensor,3)
-    float_caster = tf.cast(jpeg_decoded, dtype=tf.float32)
-    dims_expander = tf.expand_dims(float_caster, 0)
-    resizer = tf.image.resize_images(dims_expander, [224, 224], method=tf.image.ResizeMethod.NEAREST_NEIGHBOR)
+    image = tf.image.decode_png(tensor, 3)
+    image = tf.cast(image, dtype=tf.float32)
+
+    output_width = 224
+    output_height = 224
+    resize_min = 256
 
-    slice_red = tf.slice(resizer, [0, 0, 0, 0], [1, 224, 224, 1])
-    slice_green = tf.slice(resizer, [0, 0, 0, 1], [1, 224, 224, 1])
-    slice_blue = tf.slice(resizer, [0, 0, 0, 2], [1, 224, 224, 1])
+    image = _aspect_preserving_resize(image, resize_min)
```

Comment:
Does this change impact performance significantly?

Reply:
No, it's actually a little bit better

New:
Average img/sec 564.72
50.00th percentile      6.83 ms
95.00th percentile      8.03 ms
99.00th percentile      9.80 ms
99.90th percentile      17.24 ms
Top1 accuracy   73.44%
Top5 accuracy   91.23%

Old:
Average img/sec 558.91
50.00th percentile      7.86 ms
95.00th percentile      9.26 ms
99.00th percentile      10.78 ms
99.90th percentile      16.15 ms
Top1 accuracy   66.88%
Top5 accuracy   87.66%